src/physlr-molecules.cc

```diff
@@ -0,0 +1,401 @@
+	             "  --help     display this help and exit\n";
+}
+
+void
+printGraph(const graph_t& g)
+{
+	std::cout << "U\tn" << std::endl;
+	std::string node1, node2;
+	int weight;
+	auto vertexItRange = boost::vertices(g);
+	for (auto vertexIt = vertexItRange.first; vertexIt != vertexItRange.second; ++vertexIt) {
+		node1 = g[*vertexIt].name;
+		weight = g[*vertexIt].weight;
+		std::cout << node1 << "\t" << weight << "\n";
+	}
+	std::cout << "\nU\tV\tn" << std::endl;
+	auto edgeItRange = boost::edges(g);
+	for (auto edgeIt = edgeItRange.first; edgeIt != edgeItRange.second; ++edgeIt) {
+		weight = g[*edgeIt].weight;
+		node1 = g[boost::source(*edgeIt, g)].name;
+		node2 = g[boost::target(*edgeIt, g)].name;
+		std::cout << node1 << "\t" << node2 << "\t" << weight << "\n";
+	}
+}
+
+void
+readTSV(graph_t& g, const std::vector<std::string>& infiles, bool verbose)
+{
+	std::cerr << "Loading graph" << std::endl;
+#if _OPENMP
+	double sTime = omp_get_wtime();
+#endif
+
+	barcodeToIndex_t barcodeToIndex;
+	indexToBarcode_t indexToBarcode;
+	for (auto& infile : infiles) {
+		infile == "-" ? "/dev/stdin" : infile;
+		vertex_t u;
+		edge_t E;
+		std::ifstream infileStream(infile);
+		std::string line;
+		bool atEdges = false;
+		while (std::getline(infileStream, line)) {
+			if (line.empty() or line == "U\tn") {
+				continue;
+			}
+			if (line.empty() or line == "U\tV\tn") {
+				atEdges = true;
+				if (verbose) {
+					std::cerr << "Loaded vertices to graph ";
+#if _OPENMP
+					std::cerr << "in sec: " << omp_get_wtime() - sTime << std::endl;
+					sTime = omp_get_wtime();
+#endif
+				}
+				continue;
+			}
+
+			std::string node1, node2;
+			int weight;
+			std::istringstream ss(line);
+			ss >> node1;
+			if (!atEdges) {
+				ss >> weight;
+				u = boost::add_vertex(g);
+				g[u].name = node1;
+				g[u].weight = weight;
+				g[u].indexOriginal = u;
+				barcodeToIndex[node1] = u;
+				indexToBarcode[u] = node1;
+			} else {
+				ss >> node2 >> weight;
+				E = boost::add_edge(barcodeToIndex[node1], barcodeToIndex[node2], g).first;
+				g[E].weight = weight;
+			}
+		}
+	}
+	if (verbose) {
+		std::cerr << "Loaded edges to graph ";
+	} else {
+		std::cerr << "Loaded graph ";
+	}
+#if _OPENMP
+	std::cerr << "in sec: " << omp_get_wtime() - sTime << std::endl;
+	sTime = omp_get_wtime();
+#endif
+	std::cerr << "Memory usage: " << double(memory_usage()) / double(1048576) << "GB" << std::endl;
+}
+
+/* Generate a molecule separated graph (molSepG) using component/community information from
+molecule separation (vecVertexToComponent). The input graph (inG) is the barcode overlap graph or a
+molecule separated graph from the previous round of molecule separation.*/
+void
+componentsToNewGraph(
+    const graph_t& inG,
+    graph_t& molSepG,
+    vecVertexToComponent_t& vecVertexToComponent)
+{
+	barcodeToIndex_t molSepGBarcodeToIndex;
+#if _OPENMP
+	double sTime = omp_get_wtime();
+#endif
+	for (size_t i = 0; i < vecVertexToComponent.size(); i++) {
```

Comment:
```suggestion
	for (size_t i = 0; i < vecVertexToComponent.size(); ++i) {
```
Always use preincrement `++x` and avoid postincrement `x++`.

Reply:
Done.